kotlinx-coroutines-core/common/src/channels/BufferedChannel.kt

```diff
@@ -0,0 +1,2153 @@
+private val SUSPEND_NO_WAITER = Symbol("SUSPEND_NO_WAITER")
+private val FAILED = Symbol("FAILED")
+
+/*
+  Internal results for [BufferedChannel.updateCellSend]
+ */
+private const val RESULT_RENDEZVOUS = 0
+private const val RESULT_BUFFERED = 1
+private const val RESULT_SUSPEND = 2
+private const val RESULT_SUSPEND_NO_WAITER = 3
+private const val RESULT_CLOSED = 4
+private const val RESULT_FAILED = 5
+
+/**
+ * Special value for [BufferedChannel.BufferedChannelIterator.receiveResult]
+ * that indicates the absence of pre-received result.
+ */
+private val NO_RECEIVE_RESULT = Symbol("NO_RECEIVE_RESULT")
+
+
+/*
+  The channel close statuses. The transition scheme is the following:
+    +--------+   +----------------------+   +-----------+
+    | ACTIVE |-->| CANCELLATION_STARTED |-->| CANCELLED |
+    +--------+   +----------------------+   +-----------+
+        |                                         ^
+        |             +--------+                  |
+        +------------>| CLOSED |------------------+
+                      +--------+
+  We need `CANCELLATION_STARTED` to synchronize
+  concurrent closing and cancellation.
+ */
+private const val CLOSE_STATUS_ACTIVE = 0
+private const val CLOSE_STATUS_CANCELLATION_STARTED = 1
+private const val CLOSE_STATUS_CLOSED = 2
+private const val CLOSE_STATUS_CANCELLED = 3
+
+/*
+  The `senders` counter and the channel close status
+  are stored in a single 64-bit register to save the space
+  and reduce the number of reads in sending operations.
+  The code below encapsulates the required bit arithmetics.
+ */
+private const val CLOSE_STATUS_SHIFT = 60
+private const val COUNTER_MASK = (1L shl CLOSE_STATUS_SHIFT) - 1
+private inline val Long.counter get() = this and COUNTER_MASK
+private inline val Long.closeStatus: Int get() = (this shr CLOSE_STATUS_SHIFT).toInt()
+private fun constructSendersAndCloseStatus(counter: Long, closeStatus: Int): Long =
+    (closeStatus.toLong() shl CLOSE_STATUS_SHIFT) + counter
+
+/*
+  As [BufferedChannel.invokeOnClose] can be invoked concurrently
+  with channel closing, we have to synchronize them. These two
+  markers help with the synchronization. The resulting
+  [BufferedChannel.closeHandler] state diagram is presented below:
+
+    +------+ install handler +---------+  close  +---------+
+    | null |---------------->| handler |-------->| INVOKED |
+    +------+                 +---------+         +---------+
+       |             +--------+
+       +------------>| CLOSED |
+           close     +--------+
+ */
+private val CLOSE_HANDLER_CLOSED = Symbol("CLOSE_HANDLER_CLOSED")
+private val CLOSE_HANDLER_INVOKED = Symbol("CLOSE_HANDLER_INVOKED")
+
+/**
+ * Specifies the absence of closing cause, stored in [BufferedChannel.closeCause].
+ * When the channel is closed or cancelled without exception, this [NO_CLOSE_CAUSE]
+ * marker should be replaced with `null`.
+ */
+private val NO_CLOSE_CAUSE = Symbol("NO_CLOSE_CAUSE")
```

Comment:
This file has quite a few formatting violations, please apply the autoformatter and see if it helps

Reply:
it gets only worse 😄